src/components/EnvironmentBadge.js

```diff
@@ -1,30 +1,45 @@
 import React from 'react';
 import {Text, View} from 'react-native';
 import styles from '../styles/styles';
-import CONFIG from '../CONFIG';
 import CONST from '../CONST';
+import Environment from '../libs/Environment';
 
-const EnvironmentBadge = () => {
-    // If we are on production, don't show any badge
-    if (CONFIG.EXPENSIFY.ENVIRONMENT === CONST.ENVIRONMENT.PRODUCTION) {
-        return null;
-    }
+class EnvironmentBadge extends React.Component {
+    constructor() {
+        super();
 
-    const badgeText = CONFIG.EXPENSIFY.ENVIRONMENT;
-    let backgroundColorStyle = styles.badgeDanger;
+        this.state = {
+            environment: CONST.ENVIRONMENT.PRODUCTION,
+        };
+    }
 
-    if (CONFIG.EXPENSIFY.ENVIRONMENT === CONST.ENVIRONMENT.STAGING) {
-        backgroundColorStyle = styles.badgeSuccess;
+    componentDidMount() {
+        Environment.getEnvironment()
+            .then((environment) => {
+                this.setState({environment});
+            });
     }
 
-    return (
-        <View style={[styles.badge, backgroundColorStyle, styles.ml2]}>
-            <Text style={styles.badgeText}>
-                {badgeText}
-            </Text>
-        </View>
-    );
-};
+    render() {
+        // If we are on production, don't show any badge
+        if (this.state.environment === CONST.ENVIRONMENT.PRODUCTION) {
+            return null;
+        }
+
+        let backgroundColorStyle = styles.badgeDanger;
+        if (this.state.environment === CONST.ENVIRONMENT.STAGING) {
+            backgroundColorStyle = styles.badgeSuccess;
+        }
+
+        return (
+            <View style={[styles.badge, backgroundColorStyle, styles.ml2]}>
+                <Text style={styles.badgeText}>
+                    {this.state.environment}
+                </Text>
+            </View>
+        );
+    }
+}
 
 EnvironmentBadge.displayName = 'EnvironmentBadge';
```

Comment:
NAB but `displayName` is not needed for class components. Maybe it's simpler to just always have it than to have to remember that distinction though, so NAB.

Reply:
Ah ok, I saw it specified in a few other classes so wasn't sure. I'm fine with removing it if it isn't necessary, is there any documentation explaining why it isn't necessary for class components?

Reply:
I think we should always remove it for class components. Having it there on class components is an unnecessary point of possible human error.

Reply:
So feel free to remove it from any others that you see :)

Reply:
Sounds good, will do!